database-commons/src/main/java/io/cdap/plugin/db/connector/AbstractDBSpecificConnector.java

```diff
@@ -44,15 +44,18 @@
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * An Abstract DB Specific Connector those specific DB connectors can inherits
+ *
  * @param <T> the Record type that specific DB Record Reader may return while sample the data with InputFormat
  */
 public abstract class AbstractDBSpecificConnector<T extends DBWritable> extends AbstractDBConnector
   implements BatchConnector<LongWritable, T> {
 
   private final AbstractDBConnectorConfig config;
+  protected String sessionID;
```

Comment:
it usually a bad idea to have stateful variable like this, especially when it is not private, as there is no clearly defined contract around when it is set, how it should be used, etc. For example, right now it gets set in getInputFormatProvider() method. If a subclass tries to use this before that, it will be null, which may not be expected. If somebody changes the order of when methods are called, they can unintentionally break the logic.

Instead, it is better to pass this into the methods that require it. In this particular case, it looks like it only needs to be passed into the getSchemaReader() method and getStratifiedQuery() method.